src/adapter/src/coord/timestamp_oracle/postgres_oracle.rs

```diff
@@ -0,0 +1,399 @@
+// carefully tuned to hit the primary index, so the cost-based optimizer doesn't
+// really get us anything. OTOH, the background jobs that crdb creates to
+// collect these stats fill up the jobs table (slowing down all sorts of
+// things).
+//
+// WIP: Do we need this for the postgres oracle? This is originally from
+// persist.
+const SCHEMA: &str = "
+CREATE TABLE IF NOT EXISTS timestamp_oracle (
+    timeline text NOT NULL,
+    read_ts bigint NOT NULL,
+    write_ts bigint NOT NULL,
+    PRIMARY KEY(timeline)
+) WITH (sql_stats_automatic_collection_enabled = false);
+";
+
+/// A [`TimestampOracle`] backed by "Postgres".
+pub struct PostgresTimestampOracle<N>
+where
+    N: GenericNowFn<Timestamp>,
+{
+    timeline: String,
+    next: N,
+    postgres_client: Arc<PostgresClient>,
+    metrics: Arc<Metrics>,
+}
+
+/// Configuration to connect to a Postgres-backed implementation of
+/// [`TimestampOracle`].
+#[derive(Clone, Debug)]
+pub struct PostgresTimestampOracleConfig {
+    url: String,
+    metrics: Arc<Metrics>,
+
+    /// The maximum size of the connection pool to Postgres/CRDB.
+    pub connection_pool_max_size: usize,
+
+    /// The minimum TTL of a connection to Postgres/CRDB before it is
+    /// proactively terminated. Connections are routinely culled to balance load
+    /// against the downstream database.
+    pub connection_pool_ttl: Duration,
+
+    /// The minimum time between TTLing connections to Postgres/CRDB. This delay
+    /// is used to stagger reconnections to avoid stampedes and high tail
+    /// latencies. This value should be much less than `connection_pool_ttl` so
+    /// that reconnections are biased towards terminating the oldest connections
+    /// first. A value of `connection_pool_ttl / connection_pool_max_size` is
+    /// likely a good place to start so that all connections are rotated when
+    /// the pool is fully used.
+    pub connection_pool_ttl_stagger: Duration,
+
+    /// The duration to wait for a Postgres/CRDB connection to be made before
+    /// retrying.
+    pub connect_timeout: Duration,
+
+    /// The TCP user timeout for a Postgres/CRDB connection. Specifies the
+    /// amount of time that transmitted data may remain unacknowledged before
+    /// the TCP connection is forcibly closed.
+    pub tcp_user_timout: Duration,
+}
+
+impl From<PostgresTimestampOracleConfig> for PostgresClientConfig {
+    fn from(config: PostgresTimestampOracleConfig) -> Self {
+        let metrics = config.metrics.postgres_client.clone();
+        PostgresClientConfig::new(config.url.clone(), Arc::new(config), metrics)
+    }
+}
+
+impl PostgresTimestampOracleConfig {
+    /// Returns a new instance of [`PostgresTimestampOracleConfig`] with default tuning.
+    pub fn new(url: &str, metrics: Arc<Metrics>) -> Self {
+        PostgresTimestampOracleConfig {
+            url: url.to_string(),
+            metrics,
+            // NOTE: These defaults are taken as is from the persist Consensus
```

Comment:
For sure it doesn't have to be this PR, but we 100% need to make them dynamic here too. Ideally before turning this on in prod.

Reply:
Ping! Can we make this at least a TODO or something, to try to help us not forget

Reply:
will do!